krun.py

```diff
@@ -172,6 +160,15 @@ def jobs_until_eta_known(self):
     def add_job(self, job):
         self.work_deque.append(job)
 
+    def remove_job_by_key(self, key):
+        job_to_remove = None
+        for job in self.work_deque:
+            if job.key == key:
+                job_to_remove = job
+                break
+        if job_to_remove:
+            self.work_deque.remove(job_to_remove)
```

Comment:
Should there be an error case if the job was not found? You can use an `else` on the `for` loop. One of Python's curious features...